ogre/src/OgreCamera.cc

```diff
@@ -216,10 +216,11 @@ VisualPtr OgreCamera::VisualAt(const ignition::math::Vector2i
     }
   }
 
-  float ratio = screenScalingFactor();
+  float xScale, YScale;
+  screenScalingFactor(xScale, YScale);
```

Comment:
Can we cache this value instead of computing it every time the user clicks the screen? I assume this wouldn't change unless the user changes some display settings, right?

Reply:
The user could also move the window to a different screen... I think it is choosing between a little runtime overhead or very complex event-reacting code `ifdef`d for each OS or display system... But maybe I'm wrong :)